### PR TITLE
Improve the format of execution time

### PR DESCRIPTION
**Pull request by Amrysliu**
Related: https://github.com/os-autoinst/openQA/pull/2654#issuecomment-575066593

### CODE REVIEW

**Comment by Amrysliu** [comment]
Here is how the page shows
execution_time

**Comment by coolo** [comment]
That additional text adds a lot of noise. Guess the time speaks for itself

**Comment by Amrysliu** [comment]
> That additional text adds a lot of noise. Guess the time speaks for itself

Do you mean I should remove the `Execution time:`, just show the time?

File: lib/OpenQA/Utils.pm [hunk 1]
```diff
@@ -826,6 +827,20 @@ sub read_test_modules {
...
+            my %replace_hash = (
+                days    => 'd',
+                hours   => 'h',
+                minutes => 'm',
+                seconds => 's',
+            );
+            $execution_time = Time::Seconds->new($module_details->{execution_time})->pretty;
+            for my $key (keys %replace_hash) {
+                $execution_time =~ s/$key/$replace_hash{$key}/g;
+            }
```
Comment (kraih) [inline lib/OpenQA/Utils.pm]: Just writing out the four regexes would be shorter and much faster.

File: t/api/04-jobs.t [hunk 2]
```diff
@@ -1206,10 +1206,10 @@ subtest 'create job failed when PUBLISH_HDD_1 is invalid' => sub {
 
 subtest 'show job modules execution time' => sub {
     my %modules_execution_time = (
-        aplay              => 146,
-        consoletest_finish => 164,
-        gnucash            => 187,
-        installer_timezone => 34
+        aplay              => '2 m 26 s',
```
Comment (kraih) [inline t/api/04-jobs.t]: Wouldn't `2m 26s` look better without the whitespace?

Comment (Martchus) [reply t/api/04-jobs.t]: Seriously? I find the white space more clean and consistent. You would have a white space if you'd spell out the abbreviations for minutes and seconds.

Comment (Martchus) [reply t/api/04-jobs.t]: And common pages like Wikipedia also put a white space between the number and its unit, e.g. https://en.wikipedia.org/wiki/Metre

Comment (kraih) [reply t/api/04-jobs.t]: Looks like we need a vote. 😁

Comment (Amrysliu) [reply t/api/04-jobs.t]: According the result of voting, I remove the space :)

File: t/16-utils.t [hunk 3]
```diff
@@ -365,6 +365,19 @@ subtest 'project directory functions' => sub {
     is imagesdir(), '/tmp/test/openqa/images',      'right directory';
 };
 
+subtest 'change_sec_to_word' => sub {
+    is change_sec_to_word(), undef, 'do pass parameter';
+    is change_sec_to_word(1.2),    undef,           'treat float as invalid parameter';
+    is change_sec_to_word('test'), undef,           'treat string as invalid parameter';
+    is change_sec_to_word(10),     '10s',           'correctly convert';
```
Comment (kraih) [inline t/16-utils.t]: That should probably say `correctly converted`.

File: t/api/04-jobs.t (hunk 2 above)
Comment (Martchus) [reply t/api/04-jobs.t]: Not that this means not following conventions by the SI: https://en.wikipedia.org/wiki/Space_(punctuation)#Unit_symbols_and_numbers, https://www.bipm.org/en/publications/si-brochure/section5-3.html

Comment (Amrysliu) [reply t/api/04-jobs.t]: I just think this with no space look nicer.  And there is same situation in the OS, such as
```
liuxiaojing@linux-90dc:~> time ls
...
real    0m0.006s
user    0m0.002s
sys     0m0.004s
``` 
If there is no severe effect, maybe we could remove the space.

**Comment by okurz** [comment]
https://build.opensuse.org/package/live_build_log/devel:openQA:TestGithub:PR-2725/openQA/openSUSE_Tumbleweed/x86_64 shows failures:

```
[  878s]     #   Failed test 'last_updated initialized on creation'
[  878s]     #   at ./t/21-needles.t line 82.
[  878s]     #          got: '2020-02-05T13:23:57'
[  878s]     #     expected: '2020-02-05T13:23:56'
[  878s]     # Looks like you failed 1 test of 9.
[  878s] 
[  878s] #   Failed test 'handling of last update'
[  878s] #   at ./t/21-needles.t line 118.
[  878s] # Looks like you failed 1 test of 14.
[  879s] ./t/21-needles.t ................................. 
…
[  955s] # Calling ' unshare -r -n script/openqa-clone-custom-git-refspec https://github.com/user/repo/pull/9128 https://openqa.opensuse.org/t1234'
[  955s] not ok 22 - short test URLs are supported the same # TODO not implemented
[  955s] #   Failed (TODO) test 'short test URLs are supported the same'
[  955s] #   at ./t/40-script_openqa-clone-custom-git-refspec.t line 93.
[  955s] # STDOUT & STDERR:
[  955s] # openqa-clone-job --skip-chained-deps --within-instance https://openqa.opensuse.org t1234 _GROUP=0 TEST=my_test@user/repo#my_branch BUILD=user/repo#9128 CASEDIR=https://github.com/user/repo.git#my_branch PRODUCTDIR=repo/product NEEDLES_DIR=/my/case/dir/product/needles
[  955s] # 
[  955s] # don't match:
[  955s] # (?^:openqa-clone-job --skip-chained-deps --within-instance https://openqa.opensuse.org 1234 _GROUP=0 TEST=my_test@user/repo#my_branch BUILD=user/repo#9128 CASEDIR=https://github.com/user/repo.git#my_branch PRODUCTDIR=repo/product NEEDLES_DIR=/my/case/dir/product/needles)
[  955s] # as expected
[  955s] ok 23 - command exits successfully # TODO not implemented
[  955s] # Calling ' unshare -r -n script/openqa-clone-custom-git-refspec https://github.com/user/repo/pull/9128 https://openqa.opensuse.org/t1234,https://openqa.suse.de/t1234'
[  955s] not ok 24 - multiple short URLs from different hosts point to individual hosts # TODO not implemented
[  955s] #   Failed (TODO) test 'multiple short URLs from different hosts point to individual hosts'
[  955s] #   at ./t/40-script_openqa-clone-custom-git-refspec.t line 95.
[  955s] # STDOUT & STDERR:
[  955s] # openqa-clone-job --skip-chained-deps --within-instance https://openqa.opensuse.org t1234 _GROUP=0 TEST=my_test@user/repo#my_branch BUILD=user/repo#9128 CASEDIR=https://github.com/user/repo.git#my_branch PRODUCTDIR=repo/product NEEDLES_DIR=/my/case/dir/product/needles
[  955s] # openqa-clone-job --skip-chained-deps --within-instance https://openqa.suse.de t1234 _GROUP=0 TEST=my_test@user/repo#my_branch BUILD=user/repo#9128 CASEDIR=https://github.com/user/repo.git#my_branch PRODUCTDIR=repo/product NEEDLES_DIR=/my/case/dir/product/needles
[  955s] # 
[  955s] # don't match:
[  956s] # (?^s:openqa-clone-job --skip-chained-deps --within-instance https://openqa.opensuse.org 1234 _GROUP=0 TEST=my_test@user/repo#my_branch BUILD=user/repo#9128 CASEDIR=https://github.com/user/repo.git#my_branch PRODUCTDIR=repo/product NEEDLES_DIR=/my/case/dir/product/needles.* 1234)
[  956s] # as expected
[  956s] ok 25 - command exits successfully # TODO not implemented
[  956s] ok 26 - no (unexpected) warnings (via done_testing)
…
[ 1548s] Test Summary Report
[ 1548s] -------------------
[ 1548s] ./t/21-needles.t                               (Wstat: 256 Tests: 14 Failed: 1)
[ 1548s]   Failed test:  2
[ 1548s]   Non-zero exit status: 1
[ 1548s] ./t/40-script_openqa-clone-custom-git-refspec.t (Wstat: 0 Tests: 26 Failed: 0)
[ 1548s]   TODO passed:   23, 25
[ 1548s] Files=79, Tests=4129, 1340 wallclock secs ( 8.00 usr  2.39 sys + 1009.63 cusr 116.83 csys = 1136.85 CPU)
[ 1548s] Result: FAIL
```

These are unrelated but I do not want to merge this PR with failed tests as long as we have not addressed the test fails, at least in tickets.

**Comment by Amrysliu** [comment]
> These are unrelated but I do not want to merge this PR with failed tests as long as we have not addressed the test fails, at least in tickets.

I clicked the 'Download logfile' and did not see the error about 21-needle.t
```
[  451s] ./t/21-needles.t ................................. 
[  451s] [INFO] using database schema "tmp_ace0yScArP3QlfJi"
[  451s] [info] [pid:4431] setting database search path to tmp_ace0yScArP3QlfJi when registering Minion plugin
[  451s] ok 1
[  451s] # Subtest: handling of last update
[  451s]     ok 1 - all needles should have last_updated set
[  451s]     ok 2 - last_updated initialized on creation
[  451s]     ok 3 - last_updated not altered
[  451s]     ok 4 - t_updated still updated
...
[  451s] ok 14 - no (unexpected) warnings (via done_testing)
[  451s] 1..14
[  451s] ok
```
And the error in t/40-script_openqa-clone-custom-git-refspec.t, it says `# TODO not implemented`.

**Comment by okurz** [comment]
seems retriggered tests are fine, *sigh*.

**Pull request merged**